python/ray/serve/_private/deployment_state.py

```diff
@@ -1522,10 +1528,17 @@ def deploy(self, deployment_info: DeploymentInfo) -> bool:
                 deployment_info.target_capacity,
             )
 
+        # Only allow the deployment status to become UPSCALING or DOWNSCALING
+        # if the deployment is not currently UPDATING.
+        allow_scaling_statuses = (
+            self.curr_status_info.status is not DeploymentStatus.UPDATING
+        )
+
```

Comment:
This is necessary if a deployment that's still `UPDATING` is redeployed before it has finished `UPDATING`. It should not enter `UPSCALING`/`DOWNSCALING` until it has finished `UPDATING`.